iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi.hpp

```diff
@@ -91,16 +91,14 @@ class RouDi
     ///
     /// @note Intentionally not virtual to be able to call it in derived class
     void shutdown();
```

Comment:
Are we missing some `noexcept`s here or is this done intentionally in RouDi?
I know it's not the focus of this PR, just spotted it.

Reply:
Maybe we missed to add it to some functions. I've noticed this also in different places. If there are other findings in this PR I will add it but wouldn't do a commit just for that

Reply:
Sure. I can create an issue for that?

Reply:
@marthtz that would be great. Since manual scanning through the code base is monotonous and error prone, maybe clang tidy can help. @elfenpiff since you are our clang tidy expert, do you know if that's easily possible?
Maybe it can be done by #299

Reply:
@elBoberido @marthtz sadly it is not so easily feasible (https://clang.llvm.org/extra/clang-tidy/checks/list.html). For that we need some custom clang-tidy rule. This is what we already have: https://github.com/eclipse/iceoryx/blob/master/iceoryx_posh/.clang-tidy

But no one is using this clang-tidy file. I think we have to do a session like when we removed all errors and activated Werror. Then everyone is forced to follow the syntax guidelines in the clang-tidy file.